### PR TITLE
[V3 Mod] Remove tempban from data after unbanning

### LINKED ISSUE #2160: [V3 Mod] Tempban doesn't remove the user from settings on unban
# Command bugs

#### Command name

tempban

#### What cog is this command from?

Mod

#### What were you expecting to happen?

When the user is unbanned and then permanently banned, they remain permanently banned.

#### What actually happened?

The user is unbanned by the bot on the next loop due to their previous tempban having expired.

#### How can we reproduce this issue?

1. Tempban a user.
2. User is unbanned and rejoins.
3. User is permanently banned. Another tempban overrides the settings, so an extended tempban sidesteps the issue.
4. User is unbanned by the bot when the unban task loop runs again.

### PR DESCRIPTION
**Pull request by Zephyrkul**
### Type

- [X] Bugfix

### Description of the changes
Resolves #2160 
Also resolves another issue I found where the bot will error out when getting the guild's invites when it doesn't have the Manage Server permission.

### CODE REVIEW

File: redbot/cogs/mod/mod.py [hunk 1]
```diff
@@ -1392,23 +1395,24 @@ def __unload(self):
...
+                            await self.settings.member(member(uid, guild)).banned_until()
+                        )
+                        now = datetime.utcnow()
+                        if now > unban_time:  # Time to unban the user
+                            user = await self.bot.get_user_info(uid)
+                            queue_entry = (guild.id, user.id)
+                            self.unban_queue.append(queue_entry)
+                            try:
+                                await guild.unban(user, reason="Tempban finished")
+                                guild_tempbans.pop(i)
```
Comment (Tobotimus) [inline redbot/cogs/mod/mod.py]: If two tempbans are finished in this same `for` loop, the 2nd one will pop the wrong one off the list. After the first one is removed, elements in `guild_tempbans` are shifted meaning `i, uid` becomes misaligned with it. Best to use `.remove(uid)`, no need to enumerate (tempting I know due to extra performance of `.pop(i)`)

**Review by Tobotimus — CHANGES_REQUESTED** [review]
Thanks for this PR. Right idea, just one thing needs to be fixed.

**Comment by Zephyrkul** [comment]
Changes addressed, thanks for the sharp eye.

**Review by Tobotimus — APPROVED** [review]
Thanks for this PR @zephyrkul :heart:

**Pull request merged**